clippy_lints/src/bool_assert_comparison.rs

```diff
@@ -68,19 +73,27 @@ impl<'tcx> LateLintPass<'tcx> for BoolAssertComparison {
         let macros = ["assert_eq", "debug_assert_eq"];
         let inverted_macros = ["assert_ne", "debug_assert_ne"];
 
-        for mac in macros.iter().chain(inverted_macros.iter()) {
+        for (mac, is_eq) in macros
+            .iter()
+            .map(|el| (el, true))
+            .chain(inverted_macros.iter().map(|el| (el, false)))
+        {
             if let Some(span) = is_direct_expn_of(expr.span, mac) {
                 if let Some(args) = higher::extract_assert_macro_args(expr) {
                     if let [a, b, ..] = args[..] {
```

Comment:
Use this instead of `args[2..]`.
```suggestion
                    if let [a, b, fmt_args @ ..] = args[..] {
```